test/cets_SUITE.erl

```diff
@@ -2208,10 +2264,340 @@ join_interrupted_when_ping_crashes(Config) ->
+    %% Check that a start timestamp from a remote node is stored
+    Info = cets_discovery:system_info(Disco),
+    ?assertMatch(#{node_start_timestamps := #{Node2 := _}}, Info),
+    rpc(Peer2, erlang, disconnect_node, [Node1]),
+    receive
+        {log, ?FUNCTION_NAME, #{
+            level := warning,
+            meta := #{pid := Disco},
+            msg :=
+                {report, #{
+                    what := node_reconnects,
+                    start_time := StartTime,
+                    remote_node := Node2
+                }}
+        }} = M ->
+            ?assert(is_integer(StartTime), M)
+    after 5000 ->
+        ct:fail(timeout)
+    end.
+
+disco_nodeup_timestamp_is_updated_after_node_reconnects(Config) ->
+    logger_debug_h:start(#{id => ?FUNCTION_NAME}),
+    Node1 = node(),
+    #{ct2 := Peer2} = proplists:get_value(peers, Config),
+    #{ct2 := Node2} = proplists:get_value(nodes, Config),
+    Tab = make_name(Config),
+    {ok, _Pid1} = start(Node1, Tab),
+    {ok, _Pid2} = start(Peer2, Tab),
+    F = fun(State) ->
+        {{ok, [Node1, Node2]}, State}
+    end,
+    DiscoName = disco_name(Config),
+    Disco = start_disco(Node1, #{
+        name => DiscoName, backend_module => cets_discovery_fun, get_nodes_fn => F
+    }),
+    %% We have to start discovery server on both nodes for that feature to work
+    _Disco2 = start_disco(Node2, #{
+        name => DiscoName, backend_module => cets_discovery_fun, get_nodes_fn => F
+    }),
+    cets_discovery:add_table(Disco, Tab),
+    ok = cets_discovery:wait_for_ready(Disco, 5000),
+    %% Get an old nodeup timestamp
+    Info1 = cets_discovery:system_info(Disco),
+    #{nodeup_timestamps := #{Node2 := OldTimestamp}} = Info1,
+    rpc(Peer2, erlang, disconnect_node, [Node1]),
+    cets_test_wait:wait_until(
+        fun() ->
+            Info2 = cets_discovery:system_info(Disco),
+            #{nodeup_timestamps := #{Node2 := NewTimestamp}} = Info2,
+            NewTimestamp =/= OldTimestamp
+        end,
+        true
+    ).
+
+disco_node_start_timestamp_is_updated_after_node_restarts(Config) ->
+    logger_debug_h:start(#{id => ?FUNCTION_NAME}),
+    Node1 = node(),
+    #{ct2 := Peer2} = proplists:get_value(peers, Config),
+    #{ct2 := Node2} = proplists:get_value(nodes, Config),
+    Tab = make_name(Config),
+    {ok, _Pid1} = start(Node1, Tab),
+    {ok, _Pid2} = start(Peer2, Tab),
+    F = fun(State) ->
+        {{ok, [Node1, Node2]}, State}
+    end,
+    DiscoName = disco_name(Config),
+    Disco = start_disco(Node1, #{
+        name => DiscoName, backend_module => cets_discovery_fun, get_nodes_fn => F
+    }),
+    %% We have to start discovery server on both nodes for that feature to work
+    Disco2 = start_disco(Node2, #{
+        name => DiscoName, backend_module => cets_discovery_fun, get_nodes_fn => F
+    }),
+    cets_discovery:add_table(Disco, Tab),
+    ok = cets_discovery:wait_for_ready(Disco, 5000),
```

Comment:
Unless I missed something (it's difficult to compare manually), this chunk repeats for 3 test cases. Could you extract it to a helper?